Avoid file generation with `gradle init` when dialog is cancelled

alllex opened the pull request:
Fixes https://github.com/gradle/gradle/issues/18080

This PR is initially based on https://github.com/gradle/gradle/pull/23126. However, significant refactoring of the `Wrapper` task was required, so it was easier to start from a fresh branch.

In this PR, `init` task no longer depends on the `wrapper` task. Instead, it uses the shared logic to output the wrapper files. This makes sure, when the `init` task gets cancelled, no files are generated, including the wrapper files.

big-guy commented on platforms/software/build-init/src/testFixtures/groovy/org/gradle/buildinit/plugins/fixtures/ScriptDslFixture.groovy:
```diff
@@ -92,6 +92,12 @@ class ScriptDslFixture {
     }
 
+    void assertGradleFilesNotGenerated(TestFile parentFolder = rootDir) {
+        assert !getBuildFile(parentFolder).exists()
```
🤔 `getBuildFile(parentFolder).assertDoesNotExist()`?

big-guy commented on platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java:
```diff
@@ -509,4 +543,7 @@ private String detectType() {
     protected abstract UserInputHandler getUserInputHandler();
+
+    @Inject
+    protected abstract FileOperations getFileOperations();
```
❌ can you use the public version of this instead?

big-guy commented on platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java:
```diff
@@ -264,13 +271,40 @@ public void setupProjectLayout() {
     }
 
+    private void generateWrapper() {
+        FileResolver fileResolver = getFileOperations().getFileResolver();
+        File unixScript = fileResolver.resolve(WrapperDefaults.SCRIPT_PATH);
+        File jarFile = fileResolver.resolve(WrapperDefaults.JAR_FILE_PATH);
```
🤔 This might be the same as looking at `ProjectLayout.projectDirectory.file(...)`?

alllex replied:
Using layout seems to require unnecessary boilerplate. What about `getProject().file(...)` ?

alllex replied:
Right, it's not compatible with CC...

big-guy commented on subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/DefaultUserInputHandler.java:
```diff
@@ -136,6 +142,10 @@ public T transform(String sanitizedInput) {
 
     @Nullable
     private <T> T prompt(String prompt, Transformer<T, String> parser) {
+        if (interrupted.get()) {
+            return null;
+        }
```
🤔 is this necessary?  Doesn't this mean that the text from the additional questions won't be sent?

alllex replied:
Not necessary, but why would we continue sending input text, if we know the input handler got interrupted?

big-guy commented on platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java:
```diff
@@ -57,7 +56,6 @@ public void apply(Project project) {
                     "There is no build script or settings script",
                     new InitBuildOnlyIfSpec(buildFileDetails, settingsFileDetails, initBuild.getLogger())
                 );
-                initBuild.dependsOn(new InitBuildDependsOnCallable(buildFileDetails, settingsFileDetails));
```
🎉

big-guy reviewed (changes requested):
This should get another pair of eyes than me.

alllex commented:
@big-guy, I addressed your feedback. And since it was minor, I'll proceed with the merge.
Feel free to leave more comments afterward. I will address them in the follow-ups if required.

alllex commented:
@bot-gradle merge

bot-gradle commented:
Your PR is queued. See [the queue page](https://gradle-bot.grdev.net/queue.html) for details.

bot-gradle commented:
I've triggered [a build](https://builds.gradle.org/viewQueued.html?itemId=75946108) for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=9300c3024a4932ddc064c85c88477f3187628b39) to see all failures if there's any.

bot-gradle commented:
[Pre-tested commit build](https://builds.gradle.org/viewLog.html?buildId=75946108) failed. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=9300c3024a4932ddc064c85c88477f3187628b39) to see all build failures.

alllex commented:
@bot-gradle test without pts

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":null,"teamCityBuildHeadRef":"3bb8e6a03af241a09835337da037b511028d601d","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
I've triggered the following builds with parameters: `-DenablePredictiveTestSelection=false` for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=3bb8e6a03af241a09835337da037b511028d601d) to see all build failures.
- [PullRequestFeedback build](https://builds.gradle.org/viewQueued.html?itemId=75951366)

alllex commented:
@bot-gradle test without pts

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":null,"teamCityBuildHeadRef":"0a58a0ac4c63fba255aa9638fd347e8c2ad363e7","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
I've triggered the following builds with parameters: `-DenablePredictiveTestSelection=false` for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=0a58a0ac4c63fba255aa9638fd347e8c2ad363e7) to see all build failures.
- [PullRequestFeedback build](https://builds.gradle.org/viewQueued.html?itemId=75952018)

alllex commented:
@bot-gradle merge

bot-gradle commented:
Your PR is queued. See [the queue page](https://gradle-bot.grdev.net/queue.html) for details.

bot-gradle commented:
I've triggered [a build](https://builds.gradle.org/viewQueued.html?itemId=75960543) for you. Click [here](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.names=gitCommitId&amp;search.tags=CI&amp;search.values=e49e1fb957276bff80a2d82a071d845fa0298bdf) to see all failures if there's any.
